### PR TITLE
refactor genezio ast + typescript sdk + swift (buggy)

### PR DESCRIPTION
**Pull request by stefandarius**

### CODE REVIEW

File: src/generateSdk/astGenerator/TsAstGenerator.ts [hunk 1]
```diff
@@ -1,166 +1,262 @@
...
+            rawType: "",
+            paramType: this.mapTypesToParamType(parameter.type),
+            optional: parameter.questionToken ? true : false,
+            defaultValue: parameter.initializer ? {value: parameter.initializer.text, type: parameter.initializer.kind === 10 ? AstNodeType.StringLiteral : AstNodeType.DoubleLiteral} : undefined
           }
+          parameters.push(param);
+        }
+      }
+    }
+    //console.log("METHOD SIGNATURE:", methodSignatureCopy);
```
Comment (vladiulianbogdan) [inline src/generateSdk/astGenerator/TsAstGenerator.ts:145]: remove this commented log

File: src/generateSdk/astGenerator/TsAstGenerator.ts [hunk 2]
```diff
@@ -1,166 +1,262 @@
...
 
-    if (classDefinition === undefined) {
-      throw new Error("No class definition found");
-    } else {
-      return {
-        program: {
-          body: [classDefinition],
-          originalLanguage: "ts",
-          sourceType: SourceType.module
+    const node = typescript.createSourceFile('test.ts', fileData.toString(), typescript.ScriptTarget.ES2015, true, typescript.ScriptKind.TS);
```
Comment (vladiulianbogdan) [inline src/generateSdk/astGenerator/TsAstGenerator.ts:228]: what is this test.ts?

File: src/generateSdk/sdkGenerator/PythonSdkGenerator.ts [hunk 3]
```diff
@@ -161,6 +199,47 @@ class SdkGenerator implements SdkGeneratorInterface {
...
+    } else if (elem.type === AstNodeType.StringLiteral) {
+      return "str";
+    } else if (elem.type === AstNodeType.IntegerLiteral) {
+      return "int";
+    } else if (elem.type === AstNodeType.FloatLiteral || elem.type === AstNodeType.DoubleLiteral) {
+      return "float";
+    } else if (elem.type === AstNodeType.BooleanLiteral) {
+      return "bool";
+    } else if (elem.type === AstNodeType.AnyLiteral) {
+      return "Any";
```
Comment (vladiulianbogdan) [inline src/generateSdk/sdkGenerator/PythonSdkGenerator.ts:215]: Is there `Any` in python?

File: src/generateSdk/astGenerator/TsAstGenerator.ts (hunk 2 above)
Comment (stefandarius) [reply src/generateSdk/astGenerator/TsAstGenerator.ts:228]: `createSourceFile` requires a file name as first parameter. the file does not exist and it doesn't create it. i guess it's irelevant what we pass there, but we have to pass a string. should i change it?

File: src/generateSdk/sdkGenerator/PythonSdkGenerator.ts (hunk 3 above)
Comment (stefandarius) [reply src/generateSdk/sdkGenerator/PythonSdkGenerator.ts:215]: yes, as python introduced type hints, Any is available, but it has to be imported from `typing`

File: src/generateSdk/astGenerator/TsAstGenerator.ts (hunk 1 above)
Comment (stefandarius) [reply src/generateSdk/astGenerator/TsAstGenerator.ts:145]: done

**Pull request merged**